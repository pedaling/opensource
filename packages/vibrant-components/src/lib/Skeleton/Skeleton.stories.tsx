import type { ComponentMeta, ComponentStory } from '@storybook/react';
import { Skeleton } from './Skeleton';

export default {
  title: 'Skeleton',
  component: Skeleton,
  args: {
    width: 200,
    height: 200,
    borderRadiusLevel: 2,
  },
} as ComponentMeta<typeof Skeleton>;

export const Basic: ComponentStory<typeof Skeleton> = props => <Skeleton {...props} />;
