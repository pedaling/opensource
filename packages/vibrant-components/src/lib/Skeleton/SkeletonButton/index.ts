export { SkeletonButton } from './SkeletonButton';
export type { SkeletonButtonProps } from './SkeletonButtonProps';
