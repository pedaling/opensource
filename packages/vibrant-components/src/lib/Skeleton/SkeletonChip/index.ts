export { SkeletonChip } from './SkeletonChip';
export type { SkeletonChipProps } from './SkeletonChipProps';
