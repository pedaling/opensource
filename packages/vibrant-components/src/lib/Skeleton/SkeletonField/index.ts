export { SkeletonField } from './SkeletonField';
export type { SkeletonFieldProps } from './SkeletonFieldProps';
