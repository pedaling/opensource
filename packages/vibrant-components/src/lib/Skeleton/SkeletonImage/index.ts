export { SkeletonImage } from './SkeletonImage';
export type { SkeletonImageProps } from './SkeletonImageProps';
