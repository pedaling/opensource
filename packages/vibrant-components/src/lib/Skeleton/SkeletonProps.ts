import type { BorderSystemProps, ReactElementChild, ResponsiveValue } from '@vibrant-ui/core';
import { withVariation } from '@vibrant-ui/core';

export type SkeletonProps = Pick<BorderSystemProps, 'borderRadiusLevel'> & {
  width: ResponsiveValue<number | 'auto' | `${number}%`>;
  height: ResponsiveValue<number | 'auto' | `${number}%`>;
  maxWidth?: ResponsiveValue<number | `${number}%`>;
  children?: ReactElementChild;
};

export const withSkeletonVariation = withVariation<SkeletonProps>('Skeleton')();
