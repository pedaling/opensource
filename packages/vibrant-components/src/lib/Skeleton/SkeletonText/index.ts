export { SkeletonText } from './SkeletonText';
export type { SkeletonTextProps } from './SkeletonTextProps';
