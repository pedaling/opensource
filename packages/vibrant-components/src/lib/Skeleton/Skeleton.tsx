import { Box } from '@vibrant-ui/core';
import { Motion } from '@vibrant-ui/motion';
import { withSkeletonVariation } from './SkeletonProps';

export const Skeleton = withSkeletonVariation(props => (
  <Motion
    animation={{
      opacity: {
        from: 1,
        to: 0.4,
      },
    }}
    loop="reverse"
    duration={2400}
  >
    <Box backgroundColor="surface1" {...props} />
  </Motion>
));
