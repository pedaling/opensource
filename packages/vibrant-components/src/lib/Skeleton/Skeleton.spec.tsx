import { baseTheme } from '@vibrant-ui/theme';
import type { ReactRenderer } from '@vibrant-ui/utils/testing-web';
import { createReactRenderer } from '@vibrant-ui/utils/testing-web';
import { Skeleton } from './Skeleton';

describe('<Skeleton />', () => {
  const { render } = createReactRenderer();
  let renderer: ReactRenderer;
  let element: HTMLElement;

  describe('when basic Skeleton is rendered', () => {
    beforeEach(async () => {
      renderer = render(
        <Skeleton data-testid="skeleton" width={100} height={50} rounded="md" />
      );

      element = await renderer.findByTestId('skeleton');
    });

    it('should have correct dimensions and styles', () => {
      expect(element).toHaveStyleRule('width', '100px');
      expect(element).toHaveStyleRule('height', '50px');
      expect(element).toHaveStyleRule('background-color', baseTheme.colors.light.overlay);
    });

    it('match snapshot', () => {
      expect(renderer.container).toMatchSnapshot();
    });
  });

  describe('when Skeleton.Text is rendered', () => {
    beforeEach(async () => {
      renderer = render(
        <Skeleton.Text data-testid="skeleton-text" lines={3} typography="body1" />
      );
    });

    it('should render correct number of lines', async () => {
      const lines = await renderer.findAllByRole('generic');
      // VStack + (3 lines * 2 components per line)
      expect(lines.length).toBeGreaterThan(3);
    });

    it('match snapshot', () => {
      expect(renderer.container).toMatchSnapshot();
    });
  });

  describe('when Skeleton.Avatar is rendered', () => {
    beforeEach(async () => {
      renderer = render(
        <Skeleton.Avatar data-testid="skeleton-avatar" size="md" />
      );
    });

    it('match snapshot', () => {
      expect(renderer.container).toMatchSnapshot();
    });
  });

  describe('when Skeleton.Button is rendered', () => {
    beforeEach(async () => {
      renderer = render(
        <Skeleton.Button data-testid="skeleton-button" size="md" />
      );
    });

    it('match snapshot', () => {
      expect(renderer.container).toMatchSnapshot();
    });
  });

  describe('when Skeleton.Image is rendered', () => {
    beforeEach(async () => {
      renderer = render(
        <Skeleton.Image data-testid="skeleton-image" width={200} ratio={1.5} />
      );
    });

    it('match snapshot', () => {
      expect(renderer.container).toMatchSnapshot();
    });
  });
}); 