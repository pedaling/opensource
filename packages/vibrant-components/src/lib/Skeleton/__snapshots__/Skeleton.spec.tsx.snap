// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Skeleton /> when Skeleton.Avatar is rendered match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  background-color: #0000001a;
  border-radius: 10000px;
  width: 30px;
  height: 30px;
}

<div>
  <div
    class="emotion-0"
  />
</div>
`;

exports[`<Skeleton /> when Skeleton.Button is rendered match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  background-color: #0000001a;
  border-radius: 4px;
  padding-top: 10px;
  padding-bottom: 10px;
  width: 72px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  height: 18px;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    />
  </div>
</div>
`;

exports[`<Skeleton /> when Skeleton.Image is rendered match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 200px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 100%;
  height: 0;
  padding-bottom: 66.66666666666666%;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  right: 0;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  background-color: #0000001a;
  width: 100%;
  height: 100%;
  border-radius: 4px;
}

<div>
  <div
    class="emotion-0"
    data-testid="ratio"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      >
        <div
          class="emotion-3"
          data-testid="skeleton-image"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<Skeleton /> when Skeleton.Text is rendered match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 100%;
  -webkit-box-pack: start;
  -ms-flex-pack: start;
  -webkit-justify-content: flex-start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 100%;
  height: 20px;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  background-color: #0000001a;
  height: 16px;
  border-radius: 4px;
}

.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  width: 40%;
  height: 20px;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

<div>
  <div
    class="emotion-0"
  >
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      />
    </div>
    <div
      class="emotion-1"
    >
      <div
        class="emotion-2"
      />
    </div>
    <div
      class="emotion-5"
    >
      <div
        class="emotion-2"
      />
    </div>
  </div>
</div>
`;

exports[`<Skeleton /> when basic Skeleton is rendered match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  box-sizing: border-box;
  position: relative;
  background-color: #0000000d;
  width: 100px;
  height: 50px;
  border-radius: 8px;
}

<div>
  <div
    class="emotion-0"
    data-testid="skeleton"
  />
</div>
`;
